rundk/main.go

```diff
@@ -0,0 +1,192 @@
+
+	// Setup kubeconfig
+	if useLocalKubeconfig {
+		kubeconfig := os.Getenv("KUBECONFIG")
+		// If KUBECONFIG is not empty, also mount the kubeconfig files to the
+		// container
+		if kubeconfig != "" {
+			envs["KUBECONFIG"] = kubeconfig
+			for _, f := range strings.Split(kubeconfig, string(os.PathListSeparator)) {
+				if _, err := os.Stat(f); !os.IsNotExist(err) {
+					cleanup := cmd.CopyAndAddMount("bind", tmpDir, f, f)
+					builtUpDefers = append(builtUpDefers, cleanup)
+				}
+			}
+		}
+		// Copy and mount for default kube context to be available
+		defaultKubeconfigPath := path.Join(os.Getenv("HOME"), ".kube")
+		if _, err := os.Stat(defaultKubeconfigPath); !os.IsNotExist(err) {
+			cleanup := cmd.CopyAndAddMount("bind", tmpDir, path.Join(os.Getenv("HOME"), ".kube"), "/root/.kube")
+			builtUpDefers = append(builtUpDefers, cleanup)
+		}
+	}
+
+	// Setup args to promote env vars from local to the Docker container
+	if err = envs.PromoteFromEnv(mandatoryEnvVars...); err != nil {
+		log.Fatal("Missing mandatory argument: ", err)
+	}
+	envs.PromoteFromEnv(optionalEnvVars...) // Optional, so don't check error
+
+	// Setup args to mount the required files and directories on the host
+	// machine to the container
+	for _, m := range mounts {
+		sourceAndTarget := strings.Split(m, ":")
+		if len(sourceAndTarget) != 2 {
+			log.Fatalf("The mount string %q must be in the format of [source:target]", m)
+		}
+		cmd.AddMount("bind", sourceAndTarget[0], sourceAndTarget[1])
+	}
+
+	// Copy and mount source code root dir
+	repoRoot, err := helpers.GetRootDir()
+	if err != nil {
+		log.Fatal("Error getting the repo's root directory: ", err)
+	}
+	cleanup := cmd.CopyAndAddMount("bind", tmpDir, repoRoot, repoRoot)
+	builtUpDefers = append(builtUpDefers, cleanup)
+
+	// Setup logging
+	cmd.LogFile = path.Join(tmpDir, "build-log.txt")
+	log.Print("Logging to ", cmd.LogFile)
+
+	// Setup and mount ARTIFACTS directory
+	extArtifacts := os.Getenv("ARTIFACTS")
+	if extArtifacts == "" {
+		log.Print("Setting local ARTIFACTS directory to ", tmpDir)
+		extArtifacts = tmpDir
+	}
+	cmd.AddMount("bind", extArtifacts, extArtifacts)
+	envs["ARTIFACTS"] = extArtifacts
+	builtUpDefers = append(builtUpDefers, func() {
+		log.Print("💡 Artifacts can be found at ", extArtifacts)
+	})
+	cmd.AddEnv(envs)
+
+	wd, err := os.Getwd()
+	if err != nil {
+		log.Fatal("Error getting the working directory: ", err)
+	}
+	// Set starting directory to be the same as the current working directory.
+	cmd.AddArgs("-w=" + wd)
+
+	return cmd, func() {
+		for _, ff := range builtUpDefers {
+			if ff != nil {
+				ff()
+			}
+		}
+	}
+}
+
+func run(cmd interactive.Docker, commandAndArgsOpt ...string) error {
+	// Finally add the image then command to run (if any)
+	cmd.AddArgs(image)
+	log.Println(cmd)
+	if len(commandAndArgsOpt) != 0 {
+		cmd.AddArgs(entrypoint)
+		cmd.AddArgs(commandAndArgsOpt...)
+		log.Println("👆 Starting in 3 seconds, ^C to abort!")
+	}
+
+	time.Sleep(time.Second * 3)
```

Comment:
Nit: Use a unique reference for the wait time. It will simplify any potential future change which can diverge these values

Reply:
Done.